java/client/src/org/openqa/selenium/net/PortProber.java

```diff
@@ -156,4 +157,22 @@ public static boolean pollPort(int port, int timeout, TimeUnit unit) {
 
     return false;
   }
+
+  public static void waitForPortUp(int port, int timeout, TimeUnit unit) {
+    long end = System.currentTimeMillis() + unit.toMillis(timeout);
+    while (System.currentTimeMillis() < end) {
+      try {
+        Socket socket = new Socket();
+        socket.connect(new InetSocketAddress("localhost", port), 1000);
```

Comment:
@lukeis The `pollPort` function checks if a server can be created on that port, where as this checks if the server is running so that it can be connected to. Or are you referring to some other `pollPort` function that I missed?